### ISSUE TITLE
cant use parcel on android (Termux)

### ISSUE THREAD

**Issue by ternenges** [parcel-bundler/parcel#8068]
# 🐛 bug report

## 🎛 Configuration (.babelrc, package.json, cli command)

```js
package.json {
{
   "name": "mc",
   "version": "1.0.0",
   "description": "personal stuff",
   "main": "bundle.js",
   "src": "src/index.html",
   "private": true,
   "scripts": {
      "start": "clear && yarn dev",
      "dev": "parcel public/index.html",
      "build": "parcel build"
   },
   "author": "ternenge",
   "license": "ISC",
   "devDependencies": {
      "@parcel/transformer-less": "^2.5.0",
      "parcel": "^2.5.0",
      "parcel-transformer-svelte3-plus": "^0.2.8",
      "svelte-preprocess": "^4.10.6"
   },
   "dependencies": {
      "@babel/polyfill": "^7.12.1",
      "eruda-dom": "^2.0.0",
      "svelte": "^3.48.0"
   }
}

}
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.5.0
| Node             | 17.9.0 and 16.14.2-1
| npm/Yarn         | 8.5.5
| Operating System | android

**Comment by mischnic** [comment]
What error message do you get?

**Comment by ternenges** [comment]
during installation I get a make buid error about node-gyp-build. I then installed Ubuntu tried again and it worked fine. the problem here is that nodejs-lts and nodejs-current versions on termux are 1 patch behind and nvm is not supported on termux. currently LTS is 16.14.2-1 while release is 16.15 and current is 17.9.0-1 while release is 18.*.*.
on Ubuntu the versions are up to date.

**Comment by ternenges** [comment]
after installing with Ubuntu, I tried starting but I got this error about module not found some were in the source maps where it checks for the process.platform and arch. I looked at the index.js file, I found out that android wasn't included and when it hits else(default) something like *-android-arch64-*.node, that files is'nt present.

**Comment by mikezks** [comment]
Does it help to install `pkg install binutils`?

**Comment by daforster** [comment]
I just faced the same issue and ˋbinutilsˋ (together with ˋcmakeˋ and ˋbuild-essentialˋ) made it work. Many thanks for your advice, @mikezks

**Comment by ternenges** [comment]
I have all of those installed yet i het this 
$ parcel build www/index.html
Error: Cannot find module './artifacts/index.android-arm64.node'
Require stack:                                             - /data/data/com.termux/files/home/projects/editor/node_modules/@parcel/source-map/parcel_sourcemap_node/index.js     - /data/data/com.termux/files/home/projects/editor/node_modules/@parcel/source-map/dist/node.js                       - /data/data/com.termux/files/home/projects/editor/node_modules/@parcel/utils/lib/index.js                            - /data/data/com.termux/files/home/projects/editor/node_modules/@parcel/core/lib/public/Config.js                     - /data/data/com.termux/files/home/projects/editor/node_modules/@parcel/core/lib/utils.js                             - /data/data/com.termux/files/home/projects/editor/node_modules/@parcel/core/lib/public/Environment.js                - /data/data/com.termux/files/home/projects/editor/node_modules/@parcel/core/lib/public/Asset.js                      - /data/data/com.termux/files/home/projects/editor/node_modules/@parcel/core/lib/Parcel.js                            - /data/data/com.termux/files/home/projects/editor/node_modules/@parcel/core/lib/index.js
- /data/data/com.termux/files/home/projects/editor/node_modules/parcel/lib/cli.js
- /data/data/com.termux/files/home/projects/editor/node_modules/parcel/lib/bin.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (/data/data/com.termux/files/home/projects/editor/node_modules/v8-compile-cache/v8-compile-cache.js:159:20)
    at Object.<anonymous> (/data/data/com.termux/files/home/projects/editor/node_modules/@parcel/source-map/parcel_sourcemap_node/index.js:15:18)
    at Module._compile (/data/data/com.termux/files/home/projects/editor/node_modules/v8-compile-cache/v8-compile-cache.js:192:30)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/data/data/com.termux/files/home/projects/editor/node_modules/@parcel/source-map/parcel_sourcemap_node/index.js',
    '/data/data/com.termux/files/home/projects/editor/node_modules/@parcel/source-map/dist/node.js',
    '/data/data/com.termux/files/home/projects/editor/node_modules/@parcel/utils/lib/index.js',
    '/data/data/com.termux/files/home/projects/editor/node_modules/@parcel/core/lib/public/Config.js',
    '/data/data/com.termux/files/home/projects/editor/node_modules/@parcel/core/lib/utils.js',
    '/data/data/com.termux/files/home/projects/editor/node_modules/@parcel/core/lib/public/Environment.js',
    '/data/data/com.termux/files/home/projects/editor/node_modules/@parcel/core/lib/public/Asset.js',
    '/data/data/com.termux/files/home/projects/editor/node_modules/@parcel/core/lib/Parcel.js',
    '/data/data/com.termux/files/home/projects/editor/node_modules/@parcel/core/lib/index.js',
    '/data/data/com.termux/files/home/projects/editor/node_modules/parcel/lib/cli.js',
    '/data/data/com.termux/files/home/projects/editor/node_modules/parcel/lib/bin.js'
  ]
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by ternenges** [comment]
but parcel-bundler works fine

**Comment by mischnic** [comment]
There are no prebuilt binaries for android: https://unpkg.com/browse/@parcel/source-map@2.0.5/parcel_sourcemap_node/artifacts/

But we already publish Webassembly binaries as part of that package, so we should just fall back in this situation.

**Comment by komanton** [comment]
@mikezks @daforster Could you please help with this libs: ˋbinutilsˋ (together with ˋcmakeˋ and ˋbuild-essentialˋ)? How can we use them to fix the issue: 

`Error: Cannot find module './artifacts/index.android-arm64.node'`

Or should we rebuild this package @parcel/source-map with different target platform?

**Comment by RyanLarge13** [comment]
> @mikezks @daforster Could you please help with this libs: ˋbinutilsˋ (together with ˋcmakeˋ and ˋbuild-essentialˋ)? How can we use them to fix the issue:
> 
> `Error: Cannot find module './artifacts/index.android-arm64.node'`
> 
> Or should we rebuild this package @parcel/source-map with different target platform?

I am having the same issue.

**Comment by daforster** [comment]
Hi @RyanLarge13 and @komanton . That's quite long ago... I only installed these packages with apt (`apt install binutils cmake build-essential`) and it helped...so far. Because later I had faced another issue after adding react and typescript I had finally given up and switched to webpack. :see_no_evil:

**Comment by komanton** [comment]
Thanks for the answer.

At this moment, I have small progress with this issue. I just tried to turn off some how 'source-map' with help of some manual patches in artifacts. But there are a lot of other Parcel's packages which have native bindings and which are not supported by termux (android platform). 

Finally, I postponed this researches. And, for now, I am trying to use Linux inside termux (android) with help of 'proot-distro' (lightweight container based virtualization) in '--isolated' mode. I need some time to bring some results.
